### PR TITLE
Add missing dependency tags

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Adds missing dependency tags into `external_dependencies.txt` which are causing build failures.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by mssfang — APPROVED** [review]

File: sdk/monitor/azure-monitor-spring-native-test/pom.xml [hunk 1]
```diff
@@ -26,7 +26,7 @@
       <dependency>
         <groupId>org.springframework.boot</groupId>
         <artifactId>spring-boot-dependencies</artifactId>
-        <version>3.1.0</version> <!-- {x-version-update;org.springframework.boot:spring-boot-dependencies;external_dependency} -->
+        <version>2.7.13</version> <!-- {x-version-update;org.springframework.boot:spring-boot-dependencies;external_dependency} -->
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-spring-native-test/pom.xml:29]: This seems to be a downgrade.

File: sdk/monitor/azure-monitor-spring-native/pom.xml [hunk 2]
```diff
@@ -88,7 +71,7 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-enforcer-plugin</artifactId>
-        <version>3.2.1</version>  <!-- {x-version-update;org.apache.maven.plugins:maven-enforcer-plugin;external_dependency} -->
+        <version>3.0.0-M3</version>  <!-- {x-version-update;org.apache.maven.plugins:maven-enforcer-plugin;external_dependency} -->
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-spring-native/pom.xml:74]: This is also an older version.

Comment (alzimmermsft) [reply sdk/monitor/azure-monitor-spring-native/pom.xml:75]: This can be handled in a follow-up given this is causing pipeline failures

**Pull request merged**